src/catkin_pkg/changelog_generator_vcs.py

```diff
@@ -179,14 +179,23 @@ def get_latest_tag_name(self):
         tag_name = result_describe['output']
         return tag_name
 
-    def get_log_entries(self, from_tag, to_tag, skip_merges=False):
+    def get_log_entries(self, from_tag, to_tag, skip_merges=False, only_merges=False):
         # query all hashes in the range
+        MSG_ERR_OPTIONS_MERGE_CONTRADICT = 'Both "skip_merges" and "only_merges" are set to True, which contradicts.'
         cmd = [self._executable, 'log']
         if from_tag or to_tag:
             cmd.append('%s%s' % ('%s..' % to_tag if to_tag else '', from_tag if from_tag else ''))
         cmd.append('--format=format:%H')
         if skip_merges:
-            cmd.append('--no-merges')
+            if not only_merges:
```

Comment:
Seems like you would want to raise this before wasting time with the tag list, and it would also make this section cleaner. e.g. Right after defining MSG_ERR_OPTIONS_MERGE_CONTRADICT,

```
if skip_merges and only_merges:
    raise RuntimeError(MSG_ERR_OPTIONS_MERGE_CONTRADICT)
```
allows you to omit the if/else clauses for the contradicting case.

Reply:
Makes sense. Applied the change and I verifid it still works.